Use asyncio.new_event_loop().

nascheme opened the pull request:
This replaces the use of the deprecated get_event_loop() function.  In Python 3.14, get_event_loop() raises RuntimeError if there is no current event loop.

nascheme committed:
Use asyncio.new_event_loop().

This replaces the use of the deprecated get_event_loop() function.  In
Python 3.14, get_event_loop() raises RuntimeError if there is no current
event loop.

stefanseefeld commented:
Unfortunately this seemingly simple change causes a few regressions, and thus requires a bit more work.

nascheme committed:
Merge branch 'origin/develop' into async_new_event_loop

nascheme committed:
Alternative fix for get_event_loop().

This fix is suggested by Kumar Aditya <kumaraditya@python.org>.

stefanseefeld commented:
Thanks for the update ! Do you think it's possible to augment the test coverage by adding a Python 3.14 variant ? (I can of course look into this myself, bu as you already have your fingers in this endeavor, perhaps you could add a commit with a change to one of the workflow definition files under `.github/workflows/`)

nascheme committed:
Add Python 3.12, 3.13 and 3.14 to workflows.

nascheme commented:
I added 3.12 - 3.14 to the workflows.  I think we could remove 3.8 and possibly 3.9.  Both are end-of-life now.  I'm not sure there is value in adding 3.14t (free-threaded) to the list.  I think perhaps not.

stefanseefeld commented:
There are some Windows failures, but they look unrelated to the asyncio change.
(Unfortunately I don't have access to a Windows machine, so debugging this, let alone fixing this, will be somewhat of a challenge.)

nascheme committed:
Adjust default prefix logic.

This avoids the "relative path ... may not be defined outside module"
exception on Windows.  Also, on Windows, use the "ProgramFiles"
environment variable rather than a fixed path.

nascheme committed:
Close sqlite3 connection.

When the filecache object is deallocated, close the sqlite3 connection.
This avoids a PermissionError on Windows when trying to remove temporary
folders used for tests (since the .faber/filecache file was still open).

nascheme commented:
I believe I've fixed some of the Windows CI errors.  It seems the msvc tool doesn't work with anything newer than VS 2015 and only supports x86 (32-bit), AFAICT.  I'm not going to try to fix that now.

stefanseefeld commented:
Closing this as it was just merged as https://github.com/stefanseefeld/faber/pull/43 (with a small formatting fix.)
